components/Card/index.js

```diff
@@ -120,10 +119,53 @@ export default function Card({
   items,
   alertOptions,
   alertSuccess,
+  model,
 }) {
   const [searchInput, setSearchInput] = useState("");
   const [selectedItems, setSelectedItems] = useAtom(initialItemList);
 
+  const { data: storedModel, mutate: changeModel } = useSWR(`/api/${model}`);
+
+  async function handleSaveItem(item) {
+    try {
+      await fetch(`/api/${model}`, {
+        method: "POST",
+        body: JSON.stringify(item),
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+      changeModel();
+    } catch (error) {
+      console.error(error.message);
+    }
+  }
+
+  async function handleDeleteItem(id) {
+    try {
+      await fetch(`/api/${model}/${id}`, {
+        method: "DELETE",
+      });
+      changeModel();
+    } catch (error) {
+      console.error(error.message);
+    }
+  }
+  async function handleDeleteAll(model) {
+    try {
+      await fetch(`/api/${model}`, {
+        method: "DELETE",
+      });
+      changeModel();
+    } catch (error) {
+      console.error(error.message);
+    }
+  }
+
+  useEffect(() => {
+    setSelectedItems(storedModel);
+  }, [storedModel, setSelectedItems]);
+
```

Comment:
You might not need an effect here:

https://beta.reactjs.org/learn/you-might-not-need-an-effect

Reply:
i removed jotai and so i don't need the useEffect anymore